### PR TITLE
calling reload multiple times on a has many triggers only one request

### LINKED ISSUE #2327: Calling reload on hasMany with a link the second time, sends a new request even if the first one hasn't returned

### PR DESCRIPTION
**Pull request by sly7-7**
Closes #2327

cc @igorT @wecc

### CODE REVIEW

File: tests/integration/relationships/has-many-test.js [hunk 1]
```diff
@@ -627,6 +627,43 @@ test("A hasMany relationship can be directly reloaded if it was fetched via link
...
+  env.adapter.findHasMany = function(store, record, link, relationship) {
+    count++;
+    return new Ember.RSVP.Promise(function(resolve, reject) {
+      setTimeout(function() {
+        var value = [
+          { id: 1, body: "First" },
+          { id: 2, body: "Second" }
+        ];
+        resolve(value);
+      }, 100);
```
Comment (bmac) [inline tests/integration/relationships/has-many-test.js]: Lets drop the setTimeout and instead return: 

```js
return Ember.RSVP.resolve([
           { id: 1, body: "First" },
           { id: 2, body: "Second" }
       ]);
```

File: addon/-private/system/relationships/state/has-many.js [hunk 2]
```diff
@@ -112,13 +111,13 @@ ManyRelationship.prototype.reload = function() {
...
   } else {
-    return this.store.scheduleFetchMany(this.manyArray.toArray()).then(function() {
-      //Goes away after the manyArray refactor
-      self.manyArray.set('isLoaded', true);
-      return self.manyArray;
-    });
+    this._loadingPromise = promiseManyArray(this.store.scheduleFetchMany(this.manyArray.toArray()).then(() => {
+      return this.manyArray;
+    }), 'Reload with ids');
+    return this._loadingPromise;
```
Comment (bmac) [inline addon/-private/system/relationships/state/has-many.js]: @pangratz would you mind adding a test to ensure this else case is working correctly as well?

Comment (pangratz) [inline addon/-private/system/relationships/state/has-many.js]: Kindly delegating to @sly7-7 :wink:

Comment (bmac) [inline addon/-private/system/relationships/state/has-many.js]: Oops. Sorry about that.

Comment (sly7-7) [inline addon/-private/system/relationships/state/has-many.js]: :smile: No problem, I update this today.

File: tests/integration/relationships/has-many-test.js (hunk 1 above)
Comment (sly7-7) [inline tests/integration/relationships/has-many-test.js]: I thought there must be a setTimeout in order to be "sure" the multiple reloads where not resolved. But for sure it's better if I can avoid that. Will do this :)

**Comment by sly7-7** [comment]
@bmac Done. So I have to wrap the promises inside a promiseManyArray in order to be able to check the promise state. This seems to work, but I wonder if it's the right way even if I don't think we have an other way now to check the state of Promises

**Comment by pangratz** [comment]
LGTM

**Comment by bmac** [comment]
LGTM @wecc or @igorT do either of you have the bandwidth for a review?

**Comment by wecc** [comment]
LGTM :+1:

**Comment by bmac** [comment]
Thanks @sly7-7

**Pull request merged**